src/system/SystemLayer.h

```diff
@@ -195,6 +201,42 @@ class LayerLwIP : public Layer
      */
     virtual CHIP_ERROR PostEvent(Object & aTarget, EventType aEventType, uintptr_t aArgument) = 0;
 
+    /**
+     * This posts an event / message of the specified type with the provided argument to this instance's platform-specific event
+     * queue.
+     *
+     *  @param[in] event   A object encapsulate the context of a lambda
+     *
+     *  @retval    CHIP_NO_ERROR                  On success.
+     *  @retval    CHIP_ERROR_INCORRECT_STATE     If the state of the Layer object is incorrect.
+     *  @retval    CHIP_ERROR_NO_MEMORY           If the event queue is already full.
+     *  @retval    other Platform-specific errors generated indicating the reason for failure.
+     */
+    virtual CHIP_ERROR ScheduleLambdaBridge(const LambdaBridge & event) = 0;
+
+    template <typename Lambda>
+    class LambdaProxy
+    {
+    public:
+        static void Call(const void * body) { (*static_cast<const Lambda *>(body))(); }
+    };
+
+    template <typename Lambda>
+    CHIP_ERROR ScheduleLambda(const Lambda & lambda)
+    {
+        LambdaBridge event;
+
+        // memcpy is used to move the lambda into the event queue, so it must be trivially copyable
+        static_assert(std::is_trivially_copyable<Lambda>::value);
+        static_assert(sizeof(Lambda) <= CHIP_CONFIG_LAMBDA_EVENT_SIZE);
+        static_assert(alignof(Lambda) <= CHIP_CONFIG_LAMBDA_EVENT_ALIGN);
+
+        event.LambdaProxy = &LambdaProxy<Lambda>::Call;
```

Comment:
```suggestion
        event.LambdaProxy = [](const void* body) { (*static_cast<const Lambda *>(body))(); };
```

Reply:
No, this is wrong.

`LambdaProxy` is a static function pointer, which is pointer to a memory address of a static linked function, `LambdaProxy<Lambda>::Call`. But a lambda is an dynamic object created over the heap or stack.

On the handler side, the static function pointer is still valid, holding the address of the automatically generated function. On the contract, the lambda object (or its copy), doesn't hold anything, so it is unable to call the correct function using lambda.

Reply:
The currect way can be:

`&decltype([](const void* body) { (*static_cast<const Lambda *>(body))(); })::operator()`

But C++14 doesn't allow decltype on a lambda, so unless we move to C++17, we can't do it this way.

Reply:
But lambda which doesn't capture any arguments implements (per standard) a cast operator to a function pointer. I mean, if you tested that it wouldn't work, I'm ok, but it should work if I'm not missing anything :)

https://en.cppreference.com/w/cpp/language/lambda

capture-less non-generic lambda |   |  
-- | -- | --
using F = ret(*)(params);
operator F() const noexcept;

Reply:
Let me try it.

Reply:
It works! Amazing, thank you